vanilla_dijkstra_turn_restricted.go

```diff
@@ -57,7 +61,7 @@ func (graph *Graph) VanillaTurnRestrictedShortestPath(source, target int64) (flo
 		// u ← Q.extract_min()
 		u := heap.Pop(Q).(*minHeapVertex)
 		destinationRestrictionID := int64(-1)
-		if restrictions, ok := graph.restrictions[prevNodeID]; ok {
+		if restrictions, ok := restrictions[prevNodeID]; ok {
```

Comment:
What if the source node itself is restricted? Wouldn't it better to check if `u.id` is in `restrictions` and if it is then just skip it?